cli/EAP7-1100-Add_the_ability_for_the_CLI_SSL_security_commands_to_be_able_to_obtain_a_server_certificate_from_Lets_Encrypt.adoc

```diff
@@ -0,0 +1,120 @@
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WFCORE-3447 introduced high-level security commands (ssl enable-ssl-management and ssl enable-ssl-http-server)
+for enabling one-way and two-way SSL easily.
+To generate the server certificate,
+these high-level security commands made use of key-store management operations to generate a self-signed certificate.
+Since it is now possible to obtain certificates from the Let's Encrypt certificate authority using the CLI,
+the high-level security commands should be updated so that in addition to being able to generate a self-signed server certificate,
+they can also obtain a server certificate from Let's Encrypt.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1100[EAP7-1100]
+* https://issues.jboss.org/browse/WFCORE-4227[WFCORE-4227]
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+* CLI
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+* Add parameter '--lets-encrypt' to the 'enable-ssl-http-server' and 'enable-ssl-management' security commands
+* The parameter '--lets-encrypt' should be used in combination with the '--interactive' parameter
+* After '--lets-encrypt parameter' is used user is asked for several information needed to obtain the certificate:
+** Key-store file name
+** Password
+** Account Key-store file name
+** Certificate account name
+** Contact url(s)
+** Domain name(s)
+** Alias
+** Enable SSL Mutual Authentication
+** Do you agree to Let's Encrypt terms of service?
+* Introduce '--ca-account=<name>' parameter to reuse existing --ca-account
+
+==== Example of the certificate interactive setup:
+[source,bash]
+----
+[standalone@localhost:9990 /] security enable-ssl-management --interactive --lets-encrypt
+Please provide required pieces of information to enable SSL:
+
+Key-store file name (default management.keystore): a
+Password (blank generated):
+Account Key-store file name (default accounts.keystore.jks):
+Certificate account name (default CertAuthorityAccount):
+Your domain name (must be accessible by the Let's Encrypt server at 80 & 443 ports) [example.com]: da14549f.ngrok.io
+Your contact email [admin@admin.com]: admin@da14549f.ngrok.io
+Alias (blank generated):
+Do you agree to Let's Encrypt terms of service? y/n (blank n): y
+Enable SSL Mutual Authentication y/n (blank n):
+
+SSL options:
+Certificate will be obtained from Let's Encrypt server and will be valid for 90 days
+account key store file: accounts.keystore.jks
+certification authority account name: CertAuthorityAccount
+key store file: a
+domain name: da14549f.ngrok.io
+password: GwA82e2S
+alias: alias-42723f73-ec17-4c84-9c20-160180490cf8
+Server keystore file a, certificate file a.pem and a.csr file will be generated in server configuration
```

Comment:
Just a minor note, there won't be .pem or .csr file generated in the server configuration (the certificate obtained from Let's Encrypt will be added directly to the certificate key-store).